lib/core/toolrepos/autoupdate/ih_auto_update_repositories.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+repos=(engineering image-builder)
+
+for repo in "${repos[@]}" ; do
+    cd "$IH_HOME/$repo" || exit 1
```

Comment:
This is a good effort at syncing the repos without disrupting users, but I'm not convinced this is the right strategy. I would find it very confusing if I was working on a branch and had my changes suddenly stashed and popped (possibly unsuccessfully). 

Since it's probably unusual for non-infrastructure engineers to have non-default branches of these repos checked out, I think we can simplify this. 
1. If the default branch is checked out, `git pull` 
2. Otherwise, do nothing; the user is doing some work in that repo and we can leave it up to them to switch back to the default branch and pull when they are done.

I just tested [this](https://stackoverflow.com/a/42902058/2611587) idea and it seems to be possible to merge the remote into a local branch which isn't checked out using this:
```
git fetch origin master:master
```
It doesn't work if you have `master` checked out though. I don't think there's any benefit to doing this here though.

Reply:
You're right, this makes total sense. I will change it.